src/test/java/nl/tudelft/ewi/devhub/modules/MockedGitoliteGitServerModule.java

```diff
@@ -126,7 +126,6 @@ protected void createMockedMirrorsFolder() {
 	protected void createMockedRepositoriesFolder() {
 		String repositoriesPath = repositoriesFolder.toPath().toString() + "/";
 		when(configuration.getGitoliteBaseUrl()).thenReturn(repositoriesPath);
-		when(configuration.getRepositoriesDirectory()).thenReturn(repositoriesFolder);
```

Comment:
Aha hm that's an interesting question. Because this module was never designed for mocking unit tests, it is designed to act as a git server accessed through a git client by binding some of the git server implementation to its client interfaces. I see why Mockito discourages strict stubbing for unit tests, but really Mockito was used here primarily to:

1) Allow tests to spy behaviour on some of the mocked behaviour
2) Not have to go through extending every single interface in the hierarchy ourself

Perhaps it's better to rewrite this module so that it doesn't use Mockito anymore? Because I can imagine future tests that use this module failing because not all stubs will be hit in every single tests (as obviously not every possible git server interaction is used in every test). What do you think?